components/nimbus/ios/Nimbus/Utils/Logger.swift

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
The rest of the files here are copy-pastes of stuff from Glean, which I don't feel great about doing, but in order to build as a Swift package we need to have all the necessary Swift code here under the `nimbus/ios` directory.

I would love suggestions for something better to do here @travis79 @jhugman @badboy. Is it worth trying to put these more explicitly into some sort of shared utility module that both projects can depend on?

Reply:
I wouldn't mind having a shared utility module, but I guess I'm having a hard time understanding why Nimbus would need this copied over if it just depends on Glean.  Can the Nimbus Swift Package not depend on the Glean Swift Package and just import these functions directly for some reason? Even if we are megazording, could the iOS megazord itself just pull in the Glean dependence from the Swift Package instead of having Glean as a submodule, or do we lose the advantage of compiling all the Rusty bits together by doing that?

Reply:
I'll try and make some time to check out your branch to get a better understanding of all of this today, it's still early and I may not have had enough ☕ yet to grep this...

Reply:
> Can the Nimbus Swift Package not depend on the Glean Swift Package and just import these functions directly for some reason?

My high-level impression is that these utilities are not exported as part of the public API (e.g. they're not marked `public`) so one cannot import them from the Glean Swift Package. But I'll admit I haven't tried particularly hard to see if they can be imported, I'll take another look.

Reply:
So, concrete example, if we wanted to be able to import the `Sysctl` namespace from Glean then we'd need Glean to mark this struct as `public` and commit to it being part of the public API of the module:

* https://github.com/mozilla/glean/blob/d1f57a850be8987f9d59bfeea79fb2cf3a38ea6f/glean-core/ios/Glean/Utils/Sysctl.swift#L32

/cc @badboy for visibility on the discussion as well, because I could easily be missing something.

Reply:
They are private intentionally. It's not something I would want to make part of the public API really.

Now for the logger I think it might be worth standardizing on one approach, and then yes, we can probably unify this somehow.
For Sysctl, why does Nimbus need that? For now this should only be needed in Glean-internal code, so no need to expose it further.

Reply:
> They are private intentionally. It's not something I would want to make part of the public API really.
> 
> Now for the logger I think it might be worth standardizing on one approach, and then yes, we can probably unify this somehow.
> For Sysctl, why does Nimbus need that? For now this should only be needed in Glean-internal code, so no need to expose it further.

Nimbus uses sysctl just to ensure that it grabs the device arch, mfg, and model in the same format as Glean, and it borrows the `getLocaleTag` func to ensure that targeting matches the Glean client_info.

Reply:
I recall discussing in the past of creating some sort of "values provider" that both Glean and Nimbus (and anything else that needed consistent values for device info) could use, but at the time, I think it was just simpler to borrow them.

Reply:
I see. Hm, for now we _could_ expose them, but still declare them not part of the actual Glean API. That way we Nimbus could use them, but apart from Nimbus we don't need to be cautious of brekaing consumers.

Reply:
I'm happy to take on some amount of legwork to get to a good outcome here, even if it's a PR to Glean :-)

But, @travis79 @badboy I'd like to lean on your collective wisdom to suggest the right path forward.

Reply:
I'd say let's just keep the copied code for now, and we can remove it if Glean opens these functions up publicly. I don't see either of these functions changing on the Glean side anytime soon, so I think the risk of them getting out of sync is very low.